keras/layers/recurrent.py

```diff
@@ -217,7 +218,15 @@ def call(self, x, mask=None):
                                 'argument, including the time axis. '
                                 'Found input shape at layer ' + self.name +
                                 ': ' + str(input_shape))
-        if self.stateful:
+        if self.initial_states is not None:
+            if self.stateful:
+                raise Exception('Keras does not currently support stateful Recurrent '
+                                'layers with custom initial states.')
+            if type(self.initial_states) in (list, tuple):
+                initial_states = self.initial_states
+            else:
+                initial_states = [self.initial_states]
```

Comment:
I think this is meant to be like the `set_weights` function, but just for the initial hidden state (so you could have an RNN conditioned on something). 

@lobachevzky , you should model it more after: https://github.com/braingineer/keras/blob/master/keras/engine/topology.py#L831